packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/environmentVariableWalkthrough.ts

```diff
@@ -0,0 +1,161 @@
+import inquirer from 'inquirer';
+import _ from 'lodash';
+import { $TSContext } from 'amplify-cli-core';
+import { getStoredEnvironmentVariables, setEnvironmentVariable, deleteEnvironmentVariable } from '../utils/environmentVariablesHelper';
+
+export const askEnvironmentVariableQuestions = async (context: $TSContext, resourceName: string): Promise<object> => {
```

Comment:
IMO this function should not actually modify any state. It just just return an object representing the changes such as:
```
{
  envVarsToSet: [
    {
      key: 'TEST',
      value: 'testvalue'
    },
    {
      key: 'ANOTHER',
      value: 'anothertest'
    }
  ],
  envVarsToDelete: [
    'DELETE_ME', 'ALSO_DELETE_ME'
  ]
}
```

Reply:
That's an option for sure, but requires more logics around `function Parameters` in CFn template to update at the same time. The current implementation follows `scheduleWalkthrough`'s way to update properties. Let's disscuss which options to take for this time.

Refs: https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/scheduleWalkthrough.ts#L73

Reply:
the schedule walkthrough does the entire walkthrough before writing anything to CFN though which we should follow here. It leads to better separation of the walkthrough logic from the state management logic